.github/workflows/release.yml

```diff
@@ -0,0 +1,112 @@
+# This automation is to perform release tagging and artifact uploads of the LFS objects
+
+name: Publish new release
+
+on:
+  # Allows you to run this workflow manually from the Actions tag
+  workflow_dispatch:
+    inputs:
+      repository:
+        description: 'The repo to target for release'
+        required: true
+        type: choice
+        default: o3de
+        options:
+          - o3de
+          - o3de-atom-sampleviewer
+          - o3de-multiplayersample
+          - o3de-netsoaktest
+      release_tag:
+        description: 'The release tag'
+        required: true 
+        type: string
+      release_notes_url:
+        description: 'The release notes url'
+        required: false
+        type: string
+      draft_release:
+        description: 'Stage release as draft. If unchecked, publish immediately'
+        default: true
+        required: true
+        type: boolean
+        
+run-name: Releasing ${{ inputs.repository }} for ${{ inputs.release_tag }}
+
+jobs:
+  Publish:
+    name: Deploy
+    runs-on: ubuntu-latest
+    environment: ${{ github.ref_name }} # Check environment against branch
+    
+    permissions:
+      contents: write
+    
+    steps:
+      # Checks-out the repository under $GITHUB_WORKSPACE, so the job can access it
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository_owner }}/${{ inputs.repository }}
+          ref: main
+          fetch-depth: 0
+      
+      - name: Git LFS pull
+        run: |
+          git lfs install
+          git lfs fetch
+      
+      - name: Get repo vars
+        id: vars
+        run: |
+          echo ::set-output name=latest_tag::$(git describe --tags `git rev-list --tags --max-count=1`)
+          echo ::set-output name=latest_main_sha::$(git log -n 1 --pretty=format:"%H" origin/main)
+          
+      - name: Archive repo as compressed files
+        run: |
+          git archive -o ${{ inputs.repository }}-${{ inputs.release_tag }}-lfs.zip HEAD
+          git archive -o ${{ inputs.repository }}-${{ inputs.release_tag }}-lfs.tar.gz HEAD
```

Comment:
This is cool that it includes lfs files!  what if the output archive file is over 2GB? Will GitHub reject the file?

Reply:
Good catch! It will reject it if it's over 2GB. I'll add an additional step to separate the zip file if it's over 2GB, then use a glob pattern to upload

Reply:
Added a splitter step